internal/pkg/manifest/svc.go

```diff
@@ -571,6 +576,121 @@ func sortExposedPorts(exposedPorts []ExposedPort) []ExposedPort {
 	return exposedPorts
 }
 
+// HTTPLoadBalancerTarget returns target container and target port for the ALB configuration.
+// This method should be called only when ALB config is not empty.
+func (s *LoadBalancedWebService) HTTPLoadBalancerTarget() (targetContainer string, targetPort string, err error) {
+	exposedPorts, err := s.ExposedPorts()
+	if err != nil {
+		return "", "", err
+	}
+	// Route load balancer traffic to main container by default.
+	targetContainer = aws.StringValue(s.Name)
+	targetPort = s.MainContainerPort()
+
+	rrTargetContainer := s.RoutingRule.TargetContainer
+	rrTargetPort := s.RoutingRule.TargetPort
+	if rrTargetContainer == nil && rrTargetPort == nil { // both targetPort and targetContainer are nil.
+		return
+	}
+
+	if rrTargetPort == nil { // when target_port is nil
+		if rrTargetContainer != s.Name {
+			targetContainer = aws.StringValue(rrTargetContainer)
+			targetPort = aws.StringValue(s.Sidecars[aws.StringValue(rrTargetContainer)].Port)
+		}
+		return
+	}
+
+	if rrTargetContainer == nil { // when target_container is nil
+		container, port := httpLoadBalancerTarget(exposedPorts, rrTargetPort)
+		targetPort = aws.StringValue(port)
+		if container != nil {
+			targetContainer = aws.StringValue(container)
+		}
+		return
+	}
+
+	// when both target_port and target_container are not nil
+	targetContainer = aws.StringValue(rrTargetContainer)
+	targetPort = template.StrconvUint16(aws.Uint16Value(rrTargetPort))
+
+	return
+}
+
+// HTTPLoadBalancerTarget returns target container and target port for the ALB configuration.
+// This method should be called only when ALB config is not empty.
+func (s *BackendService) HTTPLoadBalancerTarget() (targetContainer string, targetPort string, err error) {
+	exposedPorts, err := s.ExposedPorts()
+	if err != nil {
+		return "", "", err
+	}
+
+	// Route load balancer traffic to main container by default.
+	targetContainer = aws.StringValue(s.Name)
+	targetPort = s.MainContainerPort()
+
+	rrTargetContainer := s.RoutingRule.TargetContainer
+	rrTargetPort := s.RoutingRule.TargetPort
+	if rrTargetContainer == nil && rrTargetPort == nil { // both targetPort and targetContainer are nil.
+		return
+	}
+
+	if rrTargetPort == nil { // when target_port is nil
+		if rrTargetContainer != s.Name {
+			targetContainer = aws.StringValue(rrTargetContainer)
+			targetPort = aws.StringValue(s.Sidecars[aws.StringValue(rrTargetContainer)].Port)
+		}
+		return
+	}
+
+	if rrTargetContainer == nil { // when target_container is nil
+		container, port := httpLoadBalancerTarget(exposedPorts, rrTargetPort)
+		targetPort = aws.StringValue(port)
+		if container != nil {
+			targetContainer = aws.StringValue(container)
+		}
+		return
+	}
+
+	// when both target_port and target_container are not nil
+	targetContainer = aws.StringValue(rrTargetContainer)
+	targetPort = template.StrconvUint16(aws.Uint16Value(rrTargetPort))
+	return
```

Comment:
Are these exactly the same as `func (s *LoadBalancedWebService) HTTPLoadBalancerTarget()`?

Reply:
Yes it is, the only difference is how we are getting sidecars and primary container name from given `LoadBalancedWebService` or `BackendService`